src/dogstatsd/client.c

```diff
@@ -116,10 +118,22 @@ dogstatsd_client_status dogstatsd_client_metric_send(
   const char *tags_prefix = (tags_len + const_tags_len > 0) ? "|#" : "";
   const char *tags_separator = (tags_len > 0 && const_tags_len > 0) ? "," : "";
 
-  const char *format = "%s:%s|%s|@%.6f%s%s%s%s";
-  int size = snprintf(client->msg_buffer, client->msg_buffer_len, format, name,
-                      value, typestr, sample_rate, tags_prefix, tags,
-                      tags_separator, client->const_tags);
+  const char *format;
+  int size;
+  /* Omit the sample rate iff it is 1.0; a sample rate of 1.000000 causes issues
```

Comment:
Tiny type-o, `iff`. Otherwise, thanks for the comment! :)

Reply:
[You're one of today's lucky 10000](https://xkcd.com/1053/)! It's a short-hand for "if and only if" in mathematics :)